types/ramda/index.d.ts

```diff
@@ -1405,7 +1405,7 @@ declare namespace R {
          * Returns a new list by pulling every item out of it (and all its sub-arrays) and putting
          * them in a new array, depth-first.
          */
-        flatten<T>(x: ReadonlyArray<T> | ReadonlyArray<T[]> | ReadonlyArray<ReadonlyArray<T>>): T[];
+        flatten<T>(x: ReadonlyArray<T[]> | ReadonlyArray<ReadonlyArray<T>> | ReadonlyArray<T>): T[];
```

Comment:
what case does this swap solves?

Reply:
Maybe I need more tests?

Reply:
I've ran some. Current typings aren't the best, but it's not because of your changes.
Your contribution is good